<launch>
  <arg name="gui" default="true"/>

  <include file="$(find jsk_pcl_ros)/sample/sample_normal_estimation_omp.launch">
    <arg name="gui" value="false"/>
  </include>
  <arg name="MANAGER" value="/right_hand_camera/right_hand_camera_nodelet_manager" />

  <node name="normal_direction_filter"
        pkg="nodelet" type="nodelet"
        args="load jsk_pcl/NormalDirectionFilter $(arg MANAGER)"
        respawn="true">
    <remap from="~input" to="normal_estimation_omp/output"/>
    <rosparam>
      use_imu: false
      eps_angle: 0.2
      direction: [0, 0, 1]
    </rosparam>
  </node>

  <group if="$(arg gui)">
    <node name="extract_indices_normal_filtered"
          pkg="nodelet" type="nodelet"
          args="load jsk_pcl/ExtractIndices $(arg MANAGER)">
      <remap from="~input" to="normal_estimation_omp/output_with_xyz"/>
      <remap from="~indices" to="normal_direction_filter/output"/>
      <rosparam>
        keep_organized: true
      </rosparam>
    </node>

    <node name="rviz"
          pkg="rviz" type="rviz"
          args="-d $(find jsk_pcl_ros)/sample/rviz/normal_direction_filter.rviz"/>
  </group>
</launch>
